plugins/nodes/src/js/components/NodesTable-new.tsx

```diff
@@ -78,6 +78,8 @@ export default class NodesTable extends React.Component<
   NodesTableProps,
   NodesTableState
 > {
+  private regionRenderer: (data: Node) => React.ReactNode;
```

Comment:
Uhm, this makes me sad 😢 

Cant we find another way injecting the necessary data? I'd rather go with `data.map(node => injectMasterRegion(node, this.props.masterRegion))` then messing around with the table already… 🙇 

```
function injectMasterRegion(node: Node, masterRegion string) {
node.__thisisbadbuttheotherooptionisevenworse.masterRegion = masterRegion;
return node;
}
```

Reply:
can you please create a JIRA (and add a comment to the code) which finds a proper solution for this redirect?

Reply:
What about enhancing `data` so we can call `data.getMaterRegion` or even `data.isInMasterRegion`. 

I think the table API has a invariant I would like to keep: every cell gets a `data: Node` which contains all the data it needs to render. So the solution to maintain this invariant would be to pass as part of the data object that missing information IMHO.

What do you think?

Reply:
@Fabs I thought about this, but it would mean that we need to pass extra data to the constructor of the nodes at every place or that we would need to access the deprecated composite state store from within the nodes struct. 

I would like to see if we need other workarounds like this for the other tables. If we do, we might need to find a fix for the table, in general, to allow us to pass in some additional general information. If we don't we should make sure that we refactor this away into the data layer once we apply it here.